src/tls/connection.h

```diff
@@ -23,16 +23,29 @@
 
 #include "wsinstance.h"
 
+typedef enum { CLIENT, WS } DataSource;
+typedef enum { SUCCESS, PARTIAL, CLOSED, RETRY, FATAL } ConnectionResult;
+
+/* hold a block of read data that is being written */
+struct ConnectionBuffer {
+  int fd;
```

Comment:
Let's revert this bit, and use client_fd and ws_fd from Connection. Plus, add the extra commit from PR #12461 to optimize the epoll.